### PR TITLE
Add `ThatAre[Not]Abstract` to `MethodInfoSelector.cs`

### PR DESCRIPTION
**Pull request by 94sedighi**
Part of #645

## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).

### CODE REVIEW

File: Src/FluentAssertions/Types/MethodInfoSelector.cs [hunk 1]
```diff
@@ -133,6 +133,26 @@ public MethodInfoSelector ThatAreNotDecoratedWithOrInherit<TAttribute>()
         return this;
     }
 
+    /// <summary>
+    /// Only return methods that are abstract
+    /// </summary>
+    /// <returns></returns>
```
Comment (jnyrup) [inline Src/FluentAssertions/Types/MethodInfoSelector.cs:139]:
```suggestion
```

File: Src/FluentAssertions/Types/MethodInfoSelector.cs [hunk 2]
```diff
@@ -133,6 +133,26 @@ public MethodInfoSelector ThatAreNotDecoratedWithOrInherit<TAttribute>()
...
+    public MethodInfoSelector ThatAreAbstract()
+    {
+        selectedMethods = selectedMethods.Where(method => method.IsAbstract);
+        return this;
+    }
+
+    /// <summary>
+    /// Only return methods that are not abstract
+    /// </summary>
+    /// <returns></returns>
```
Comment (jnyrup) [inline Src/FluentAssertions/Types/MethodInfoSelector.cs:149]:
```suggestion
```

File: Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs [hunk 3]
```diff
@@ -273,6 +273,34 @@ public void When_selecting_methods_not_decorated_with_a_noninheritable_attribute
...
+    {
+        // Arrange
+        Type type = typeof(TestClassForMethodSelectorWithAbstractAndVirtualMethods);
+
+        // Act
+        IEnumerable<MethodInfo> methods = type.Methods().ThatAreAbstract().ToArray();
+
+        // Assert
+        int abstractMethodsCount = 3;
+        methods.Should().HaveCount(abstractMethodsCount);
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs:287]:
```suggestion
        methods.Should().HaveCount(3);
```

File: Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs [hunk 4]
```diff
@@ -273,6 +273,34 @@ public void When_selecting_methods_not_decorated_with_a_noninheritable_attribute
...
+    {
+        // Arrange
+        Type type = typeof(TestClassForMethodSelectorWithAbstractAndVirtualMethods);
+
+        // Act
+        IEnumerable<MethodInfo> methods = type.Methods().ThatAreNotAbstract().ToArray();
+
+        // Assert
+        int notAbstractMethodsCount = 7;
+        methods.Should().HaveCount(notAbstractMethodsCount);
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Types/MethodInfoSelectorSpecs.cs:301]:
```suggestion
        methods.Should().HaveCount(7);
```

**Review by 94sedighi — COMMENTED** [review]
accepted changes

File: Src/FluentAssertions/Types/MethodInfoSelector.cs [hunk 5]
```diff
@@ -133,6 +133,24 @@ public MethodInfoSelector ThatAreNotDecoratedWithOrInherit<TAttribute>()
         return this;
     }
 
+    /// <summary>
+    /// Only return methods that are abstract
+    /// </summary>
+    public MethodInfoSelector ThatAreAbstract()
+    {
+        selectedMethods = selectedMethods.Where(method => method.IsAbstract);
```
Comment (jnyrup) [inline Src/FluentAssertions/Types/MethodInfoSelector.cs:141]: For consistency let's use `IsCSharpAbstract()` here as well.

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Types/MethodInfoSelector.cs (hunk 5 above)
Comment (jnyrup) [reply Src/FluentAssertions/Types/MethodInfoSelector.cs:141]: I couldn't come up with an example as well, but that does not imply that no such example exist, that's why I wanted to go for consistency.

But let's just go ahead and assume we're right until someone comes up with a counter example.

**Review by jnyrup — APPROVED** [review]

**Pull request merged**